### PR TITLE
#26 refined root readme to better cover processing pipeline

### PR DESCRIPTION
**Pull request by IanGrimstead**
Improve description of system

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,26 +29,59 @@ assess traffic flow in different parts of the country via the internet. The imag
 available, low resolution, and do not permit people or vehicles to be individually identified. They differ from CCTV
 used for public safety and law enforcement for Automatic Number Plate Recognition (ANPR) or for monitoring traffic speed.
 
-## Architecture
+## Machine Learning Pipeline
```
Comment (lichen-dsc) [inline README.md:32]: Is it better to call 'data processing pipeline' or 'machine learning based pipeline' as ML is part of the pipeline and Fig. also includes ingestion, imputation and reporting parts?

File: README.md [hunk 2]
```diff
@@ -29,26 +29,59 @@ assess traffic flow in different parts of the country via the internet. The imag
...
+depth (snapping similar colours together) and then looking at the highest fraction of the image
+occupied by a single colour. Once this is over a threshold, we determine the image is synthetic and mark it
+as faulty. Other faults may occur due to encoding, such as:
+![](tests/test_data/failing_images/TfL-images_20200504_0240_00001.08859.jpg)
+
+Here, the camrera feed has stalled and the last "live" row has been repeated; we detect this by checking
+if the bottom row of image matches the row above (within threshold). If so, then the next row above is checked
+for a match and so on until rows no longer match or we run out of rows. If the number of matching rows is above a
+threshold, then the image is unlikely to generate useful data and hence is flagged as faulty.
+
```
Comment (lichen-dsc) [inline README.md:71]: Would it be good to put a note 'When adapting the faulty object detection part to a different country or some local cameras, it is recommended to check the local camera faulty situations which may be different from the UK.'?

**Comment by lichen-dsc** [comment]
A good level of details on the data processing pipeline is provided! Thanks. 
ps. I put two minor comments next to the changes.

File: README.md (hunk 1 above)
Comment (IanGrimstead) [reply README.md]: good point - fixed

File: README.md (hunk 2 above)
Comment (IanGrimstead) [reply README.md:71]: Ooops forgot that! Thank-you, fixed.

**Review by lichen-dsc — APPROVED** [review]

**Pull request merged**